### ISSUE TITLE
React18 compatibility

### ISSUE THREAD

**Issue by ZiedChekir** [zalify/easy-email#283]
Hello there,
I was just wondering if the Easy-email editor is compatible with React18 or is it only designed to work with React17? If it is not yet compatible with React18, are there any plans to migrate it in the future? Thank you so much!

**Comment by pushchris** [comment]
<img width="528" alt="Screenshot 2023-04-12 at 10 29 46 PM" src="https://user-images.githubusercontent.com/679974/231641715-cf906f14-910a-45a4-980a-672c411c1f40.png">
It appears there is a conflict when trying to install it while also using React 18

**Comment by Nilegfx** [comment]
> React 18 can be used directly without compatibility issues.

That is not true, even copy-pasting the example from the read me file with react 18 doesn't work I got 

```
Type '{ children: Element; compact: boolean; showSourceCode: true; }' is not assignable to type 'IntrinsicAttributes & ExtensionProps'.
  Property 'children' does not exist on type 'IntrinsicAttributes & ExtensionProps'
```

quick steps to reproduce
1. run `yarn create vite` choose react with typescript
2. follow the docs of easy-email and copy paste the example code in the docs

**Comment by m-Ryan** [comment]
It just affects type hints because react-18 React.FC removes children props.
Maybe it's time to upgrade to react-18.

**Comment by m-Ryan** [comment]
In version 4.13.0, we have updated to react-18,  please check @pushchris @Nilegfx @ZiedChekir @ghyghoo8

**Comment by parathan-velaris** [comment]
Seems like it supports React 18. I used `React 17.2.0`
where I got dependency tree error since the easy-email-core, easy-email-editor and easy-email-extension are dependent on React 18.

So, I added following to package.json

`  "overrides": {
    "easy-email-core": {
      "react": "^17.0.2",
      "react-dom": "^17.0.2"
    },
    "easy-email-editor": {
      "react": "^17.0.2",
      "react-dom": "^17.0.2"
    },
    "easy-email-extensions": {
      "react": "^17.0.2",
      "react-dom": "^17.0.2"
    }
  },`

**Comment by ZiedChekir** [comment]
I confirm it is working smoothly with react version 18. Thank you for all your effort.
